serde_macros/src/field.rs

```diff
@@ -30,47 +30,99 @@ pub fn field_rename<'a>(
             if let ast::MetaList(_, ref vals) = sa.node.value.node {
                 attr::mark_used(&sa);
                 vals.iter().fold(None, |v, mi| {
-                    if let ast::MetaNameValue(ref n, ref lit) = mi.node {
-                        if n == &"rename" || n == &dir_attr {
-                            Some(lit)
-                        } else {
-                            v
-                        }
-                    } else {
-                        v
+                    match mi.node {
```

Comment:
Could we factor this out too? Maybe into an `attr.rs` file? These syntax extension files are already so messy :)